plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnector.java

```diff
@@ -1667,24 +1273,19 @@ private String getBuildImageId(ProgressStatus progressStatus) {
         return null;
     }
 
-    @VisibleForTesting
-    <T> T parseResponseStreamAndClose(InputStream inputStream, Class<T> clazz) throws IOException, JsonParseException {
-        try (InputStream responseStream = inputStream) {
-            return JsonHelper.fromJson(responseStream,
-                                       clazz,
-                                       null,
-                                       FIRST_LETTER_LOWERCASE);
+    protected <T> T parseResponseStreamAndClose(InputStream inputStream, Class<T> clazz) throws IOException {
+        try (InputStreamReader reader = new InputStreamReader(inputStream)) {
+            return GSON.fromJson(reader, clazz);
+        } catch (JsonParseException e) {
+            throw new IOException(e.getLocalizedMessage(), e);
         }
     }
 
-    @VisibleForTesting
-    @SuppressWarnings("unchecked")
-    <T> List<T> parseResponseStreamAsListAndClose(InputStream inputStream, Type type) throws IOException, JsonParseException {
-        try (InputStream responseStream = inputStream) {
-            return (List<T>)JsonHelper.fromJson(responseStream,
-                                                List.class,
-                                                type,
-                                                FIRST_LETTER_LOWERCASE);
+    protected <T> List<T> parseResponseStreamAsListAndClose(InputStream inputStream, Type type) throws IOException {
```

Comment:
I would redesign this method as it is not typesafe and _List_ in the method name is not really necessary 
e.g. 
```java
public static <T> T parseStreamAndClose(InputStream is, TypeToken<T> tt)
```
makes more sense + it's typesafe